internal/tracker/bundle_info.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package tracker
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/remote/oci"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/hacbs-contract/ec-cli/internal/image"
+)
+
+type bundleInfo struct {
+	ref image.ImageReference
+	// Name of the Task definitions found in the bundle.
+	tasks []string
+	// Name of the Pipeline definitions found in the bundle.
+	pipelines []string
+	// Set of common Tasks found across Pipelines definitions in the bundle.
+	commonPipelineTasks commonTasksRecord
+	// Set of collection where the bundle should be tracked under.
+	collections sets.String
```

Comment:
It's odd for me to see a set here, seems to me that this is used as a enum instead.

Reply:
I think I've now realized that the purpose of this, the set threw me of. I guess, this could be a something like:

```go
const (
	PIPELINE			int = 1
	TASK				int = 2
	PIPELINE_AND_TASK	int = 3
)
```

```suggestion
	type int
```

Perhaps I'm overreacting based on the first (wrong) impression...

Reply:
It's a "private" attribute and meant to be used internally by the `bundleInfo` struct. A set simplifies the code quite a bit so I'd prefer not to move away from that.